internal/exec/terraform_clean.go

```diff
@@ -0,0 +1,432 @@
+	confirmPrompt := huh.NewConfirm().
+		Title(message).
+		Affirmative("Yes!").
+		Negative("No.").
+		Value(&confirm).WithTheme(t)
+	if err := confirmPrompt.Run(); err != nil {
+		if err == huh.ErrUserAborted {
+			return confirm, fmt.Errorf("Mission aborted")
+		}
+		return confirm, err
+	}
+
+	return confirm, nil
+}
+
+// DeletePathTerraform deletes the specified file or folder. with a checkmark or xmark
+func DeletePathTerraform(fullPath string, objectName string) error {
+	if _, err := os.Stat(fullPath); os.IsNotExist(err) {
+		xMark := lipgloss.NewStyle().Foreground(lipgloss.Color("9")).SetString("x")
+		fmt.Printf("%s Cannot delete %s: path does not exist", xMark, objectName)
+		fmt.Println()
+		return err
+	}
+	// Attempt to delete the file or folder If the path does not exist, RemoveAll returns nil (no error)
+	err := os.RemoveAll(fullPath)
+	if err != nil {
+		xMark := lipgloss.NewStyle().Foreground(lipgloss.Color("9")).SetString("x")
+		fmt.Printf("%s Error deleting %s", xMark, objectName)
+		fmt.Println()
+		return err
+	}
+
+	checkMark := lipgloss.NewStyle().Foreground(lipgloss.Color("42")).SetString("✓")
+	fmt.Printf("%s deleted %s", checkMark, objectName)
+	fmt.Println()
+	return nil
+}
+
+// confirmDeletion prompts the user for confirmation before deletion.
+func confirmDeletion(cliConfig schema.CliConfiguration) (bool, error) {
+	message := "Are you sure?"
+	confirm, err := confirmDeleteTerraformLocal(message)
+	if err != nil {
+		return false, err
+	}
+	if !confirm {
+		u.LogWarning(cliConfig, "Mission aborted.")
+		return false, nil
+	}
+	return true, nil
+}
+
+// deleteFolders handles the deletion of the specified folders and files.
+func deleteFolders(folders []Directory, relativePath string) {
+	for _, folder := range folders {
+		for _, file := range folder.Files {
+			path := filepath.ToSlash(filepath.Join(relativePath, file.Name))
+			if file.IsDir {
+				DeletePathTerraform(file.FullPath, path+"/")
+			} else {
+				DeletePathTerraform(file.FullPath, path)
+			}
+		}
+	}
+	// check if the folder is empty by using the os.ReadDir function
+	for _, folder := range folders {
+		entries, err := os.ReadDir(folder.FullPath)
+		if err == nil && len(entries) == 0 {
+			os.Remove(folder.FullPath)
+		}
+	}
+
+}
+
+// handleTFDataDir handles the deletion of the TF_DATA_DIR if specified.
+func handleTFDataDir(componentPath string, cliConfig schema.CliConfiguration) {
+	tfDataDir := os.Getenv("TF_DATA_DIR")
+	if tfDataDir == "" || tfDataDir == "." || tfDataDir == "/" || tfDataDir == "./" {
+		return
+	}
+
+	u.PrintMessage(fmt.Sprintf("Found ENV var TF_DATA_DIR=%s", tfDataDir))
+	u.PrintMessage(fmt.Sprintf("Do you want to delete the folder '%s'? (only 'yes' will be accepted to approve)\n", tfDataDir))
+	fmt.Print("Enter a value: ")
+
+	var userAnswer string
+	if _, err := fmt.Scanln(&userAnswer); err != nil {
+		u.LogWarning(cliConfig, fmt.Sprintf("Error reading input: %v", err))
+		return
+	}
+
+	if userAnswer == "yes" {
```

Comment:
Why are we not using the confirmation dialog?

Reply:
I think it shouldn't be changed . I will do